forge-game/src/main/java/forge/game/ability/effects/GoadEffect.java

```diff
@@ -40,6 +41,11 @@ public void resolve(SpellAbility sa) {
                 continue;
             }
 
+            if (ungoad) {
+                tgtC.unGoad();
+                continue;
+            }
```

Comment:
@Northmoc what do you think? is this better?

Reply:
Yes – I think that's better! 😄